cpp/oneapi/dal/backend/primitives/selection/row_partitioning_dpc.cpp

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/selection/row_partitioning.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+int constexpr preffered_sg_size = 16;
+
+sycl::nd_range<2> get_row_partitioning_range(std::int64_t row_count, std::int64_t col_count)
+{
+    sycl::range<2> global(preffered_sg_size, row_count);
+    sycl::range<2> local(preffered_sg_size, 1);
```

Comment:
Unchecked cast from std::uint64_t -> std::size_t. Suggest to create function that creates ndrange safely and put it to backend/common.hpp to make it available to everybody. There is similar function called make_multiple_nd_range_1d.

Reply:
Such function was already added, you can use it.